AsyncDisplayKit/Private/ASControlTargetAction.h

```diff
@@ -0,0 +1,29 @@
+//
+//  ASControlTargetAction.h
+//  AsyncDisplayKit
+//
+//  Copyright (c) 2014-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
+
+#import <Foundation/Foundation.h>
+
+/**
+ @abstract ASControlTargetAction stores target action pairs registered for specific ASControlNodeEvent values.
+ ASControlNode This class stores target action pairs similar to the UIControlTargetAction class.
+ */
+@interface ASControlTargetAction : NSObject
+
+/** 
+ The action to be called on the registered target.
+ */
+@property (nonatomic, readwrite, assign) SEL action;
+
+/**
+ Event handler target. The specified action will be called on this object.
+ */
+@property (nonatomic, readwrite, strong) id target;
```

Comment:
We should store this `weak` to prevent retain cycles.

It's important that we find a way to distinguish between a target-action that was created with `nil` target, so it should travel up the responder chain, and one that was created with a target but the target has deallocated. We could use `NSNull` to signify the former, but that's likely to be surprising for future developers. A better move might be to add a `BOOL createdWithNoTarget` flag to make the distinction.